### PR TITLE
feat: add configuration to run specific scripts on main thread

### PR DESCRIPTION
**Pull request by DanBeckDev**
Feature as requested by: #161

It was originally suggested that the configuration should look like this:
```js
<script>
partytown = {
  loadWithPartytown: (url) => {
     if (/* your own logic for what to load in partytown vs not */) return true;
     return false;
  }
}
</script>
```
I have created it as an array:
```js
/**
   * This array can be used to filter which script are executed via
   * Partytown and which you would like to execute on the main thread.
   * 
   * @example loadScriptsOnMainThread:['https://test.com/analytics.js']
   * // Loads the `https://test.com/analytics.js` script on the main thread 
*/
<script>
partytown = {
  loadScriptsOnMainThread: ['https://test.com/analytics.js']
}
</script>
```
This is because when the function gets serialized the function is a string and cannot execute.
If we wish to proceed with the idea above I am happy to extend the serialisation functionality but I deem it overkill myself.

### CODE REVIEW

File: src/lib/types.ts
Comment (khuezy) [inline src/lib/types.ts:435]:
```suggestion
  loadScriptsOnMainThread?: string[]
```

File: src/lib/web-worker/worker-script.ts
Comment (khuezy) [inline src/lib/web-worker/worker-script.ts:29]:
```suggestion
        if (this.type && config.loadScriptsOnMainThread) {
```

**Comment by DanBeckDev** [comment]
Thank you for the review @khuezy.
Please see my latest commit resolving the issues found above.

**Review by khuezy — APPROVED** [review]
Thanks for the PR! I'm not a code owner so I don't think my approval would be sufficient.

**Comment by DanBeckDev** [comment]
@steve8708 Any chance I could get a review, please?

**Comment by robertwatts** [comment]
Hello, thank you so much for taking this on @DanBeckDev. We want to use PartyTown on wonderbly.com but are blocked by this see. If @adamdbradley is able to take a look so this feature can go out then we would be very grateful - I'm sure this would be useful for many others as well.

**Review by mhevery — APPROVED** [review]

**Comment by garethweaver** [comment]
This is great, thanks!

**Comment by jahilldev** [comment]
Hey @DanBeckDev,

It would be really nice to be able to use regex within this array, given the following asset URL that potentially contains some hash or file versioning that's not known at runtime within the application:

```ts
// https://cdn.test/20bba583/analytics.js

loadScriptsOnMainThread: [/\/analytics\.js$/]
```

**Pull request merged**
